components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -182,8 +182,21 @@ sealed class TabListAction : BrowserAction() {
      * @property tabs the [TabSessionState]s to restore.
      * @property selectedTabId the ID of the tab to select.
      */
-    data class RestoreAction(val tabs: List<TabSessionState>, val selectedTabId: String? = null) :
-        TabListAction()
+    data class RestoreAction(
+        val tabs: List<RecoverableTab>,
+        val selectedTabId: String? = null,
+        val restoreLocation: RestoreLocation
+    ) : TabListAction()
+
+    /**
+     * Indicates what location the tabs should be restored at
+     *
+     */
+    enum class RestoreLocation {
+        BEGINNING,
+        END,
+        AT_INDEX
+    }
```

Comment:
nit: Since this is specifically for `RestoreAction`, let's move it inside the `RestoreAction {}` block (So that it will be `RestoreAction.RestoreLocation.*`.